rclcpp/src/rclcpp/callback_group.cpp

```diff
@@ -23,6 +23,13 @@ CallbackGroup::CallbackGroup(CallbackGroupType group_type)
 : type_(group_type), can_be_taken_from_(true)
 {}
 
+const std::vector<rclcpp::PublisherBase::WeakPtr> &
```

Comment:
This pattern is not safe. You need to copy **while locking** here if you plan to allow concurrent access.